### PR TITLE
chore: remove the `rc` provision in the helm release workflow now that prefect 3 is GA

### PR DESCRIPTION
**Pull request by jamiezieziula**
Resolves PLA-237

### CODE REVIEW

**Review by jimid27 — APPROVED** [review]

File: .github/workflows/helm-release.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
           echo "RELEASE_VERSION=$(date +'%Y.%-m.%-d%H%M%S')" >> $GITHUB_OUTPUT
           echo "PREFECT_VERSION=$(\
             git ls-remote --tags --refs --sort="v:refname" \
-            https://github.com/PrefectHQ/prefect.git '[!prefect-]*.*.[!rc]' | tail -n1 | sed 's/.*\///' \
+            https://github.com/PrefectHQ/prefect.git '[!prefect-]*.*.*' | tail -n1 | sed 's/.*\///' \
```
Comment (parkedwards) [inline .github/workflows/helm-release.yaml:28]: these will include rcs - wonder if we should do something like this?

```
➜ git ls-remote --tags --refs --sort="v:refname" \
            https://github.com/PrefectHQ/prefect.git '[3]*.*.[!rc]'

c40d069de2d182af586eaf7909ee77beba17fd46	refs/tags/3.0.0
```

Comment (jamiezieziula) [reply .github/workflows/helm-release.yaml:28]: we aren't cutting any more RCs, but I guess its fine to leave as is?

Comment (parkedwards) [reply .github/workflows/helm-release.yaml:28]: >we aren't cutting any more RCs, but I guess its fine to leave as is?

right, the `[!rc]` just prevents them from being returned in the tag query, so we never run the risk of bundling an rc tag here

Comment (jamiezieziula) [reply .github/workflows/helm-release.yaml:28]: i see, ill update!

**Review by parkedwards — APPROVED** [review]

**Pull request merged**